File: problem-solving/parsers/py/_03.py
def parser(f):
	n = int(f.readline())
	blocks1 = [ int(f.readline()) for _ in range(n) ]
	blocks2 = [ int(f.readline()) for _ in range(n) ]

	assert(f.readline() == "\n")

	cert = int(f.readline())
	assert(f.readline().strip() == "")

	return cert, (blocks1, blocks2,)

def verifier(cert, ans):
	return cert == ans

def error(cert, input, ans):
	print("Input:")
	print("blocks1 = " + str(input[0]))
	print("blocks2 = " + str(input[1]))
	print("")
	print("Expected:")
	print(cert)
	print("")
	print("Actual:")
	print(ans)
	print()

File: problem-solving/parsers/java/_03.java
import java.util.Scanner;

public class _03 implements Parser {
    public Object[] parser(Scanner f) {
        int n = f.nextInt();
        int[] blocks1 = Parser.parse_arr_int(f, n);
        int[] blocks2 = Parser.parse_arr_int(f, n);

        assert(f.nextLine() == "");
        assert(f.nextLine() == "");

        int cert = f.nextInt();

        return Parser.ret_parser(cert, blocks1, blocks2);
    }

    public Object run(Object[] input) {
        return Solutions.realEstatePrices((int[])input[0]);
    }

    public boolean verifier(Object cert, Object ans) {
        return (int)cert == (int)ans;
    }

    public void error(Object cert, Object[] input, Object ans) {
        System.out.println(String.format(
            "Input:\n" +
            "blocks1 = %s\n" +
            "blocks2 = %s\n" +
            "\n" +
            "Expected:\n" +
            "%d\n" +
            "\n" +
            "Actual:\n" +
            "%d\n",
            Parser.print_arr((int[])input[0]),
            Parser.print_arr((int[])input[1]),
            (int)cert,
            (int)ans
        ));
    }

}
